File: translationmap/react/timerdatasource.js
//#pragma: declaration
class TimerDataSource {
  #listeners
  #value;
  #filtered;

  constructor(initialvalue){  
      this.#listeners = []
      this.#value = initialvalue;
      globalThis.setInterval(()=>{
        this.#value = this.#value + 1; 
        console.log("adding 1 to value", this.#value);
        this.notifyListeners(); 
      }, 1000); 
  }
        
  setValue(v){ this.#value = v; this.notifyListeners(); }
  getValue() { return this.#value }
  reset() { 
    this.#value = 0; 
    this.notifyListeners(); 
   }
            
  addListener(l){
      this.#listeners.push(l)
  }

  notifyListeners(){
      console.log("number of listeners:", this.#listeners.length)
      this.#listeners.forEach(l => {
          l(this.#value);
      })
  }
}
             
//#pragma: global
const datasource[[id]] = new TimerDataSource(0);

//#pragma: setup
datasource[[id]].addListener(() => {
  this.setState({});
});

File: translation/flutter/timerdatasource.dart
//#pragma: imports
import 'dart:async';

//#pragma: declaration
class TimerDataSource extends ChangeNotifier {
  int _value = 0;
  TimerDataSource() {
    Timer.periodic(Duration(seconds: 1), (t){
        this.value = this._value + 1;
  });
  }
  set value(v) {
      _value = v;
      notifyListeners();
      
  }

  reset(argument) {
    this.value = 0;
    notifyListeners(); 
  }
  
  refresh(argument) {
    notifyListeners(); 
  }

  setValue(v) { _value = v; notifyListeners(); }
  getValue() { return _value; }
}
             
//#pragma: global
TimerDataSource datasource[[id]] = TimerDataSource();

//#pragma: setup
datasource[[id]].addListener((){
  print("new");
  setState(() {});
});